recipes/java_agent.rb

```diff
@@ -32,6 +32,7 @@
 execute 'unzip-appdynamics-java-agent' do
   cwd agent['install_dir']
   command "unzip -qqo #{node['appdynamics']['java_agent']['zip']}"
+  command "chown -R #{agent['owner']}:#{agent['group']} #{agent['install_dir']}"
```

Comment:
i added a directory resource for the entire install directory already this directory resource is immediate after the unzip, @alfredmacaraeg  are you trying to do something else?